ggez-fe/src/main.rs

```diff
@@ -164,39 +166,17 @@ impl event::EventHandler for MainState {
     }
 
     fn key_up_event(&mut self, _ctx: &mut Context, keycode: Keycode, _keymod: Mod, _repeat: bool) {
+        let controller1 = PlayerController::new(self.first_player_id);
+        let controller2 = PlayerController::new(self.second_player_id);
```

Comment:
This could be improved. We should have the controllers incorporated on the world somehow. Maybe a map `<entity_id, controller>` on the world could do the trick.

Reply:
Hum, we could also have the FE saving on the world something like: `<entity_id, input>` and then the runner iterates the player controllers, and they process the input.